Orange/widgets/utils/itemmodels.py

```diff
@@ -222,39 +239,34 @@ def __setitem__(self, i, value):
     def wrap(self, table):
         self.beginResetModel()
         self._table = table
+        rows, cols = range(len(table)), range(len(table[0]) if table else 0)
+        self._roleData = [[{} for _ in cols] for _ in rows]
```

Comment:
I don't know whether this was sarcasm or not. To be on the safe side, I'll reply. :)

It is unlikely that `PyListModel` will have 1e8 rows, but quite likely that a table will have 1e7 (=1000 x 10000) cells and take 2,9 GB - without the widget actually using other roles.

We use `PyListModel` for settings or attribute lists, which are reasonably small, so no harm done here. I support the idea of adding other roles, but setting individual cell roles will make sense only for small tables (like confusion matrices, perhaps). If larger tables define roles, they will overload the method `data` and compute the role values on the fly. I guess `_roleData` should be `None` unless set explicitly.